### PR TITLE
Add query parameter to bankSearch endpoints,

### PR DESCRIPTION
**Pull request by tsehelnyk**
Add isActive field to Bank search API, bank and bankProfiles models,
possibility to get only active data or all data, 
add migration.

### CODE REVIEW

File: opba-banking-rest-impl/src/main/java/de/adorsys/opba/tppbankingapi/service/BankService.java [hunk 1]
```diff
@@ -23,15 +23,17 @@
...
+                .filter(profile -> !onlyActive || profile.isActive())
                 .map(BankProfile.TO_BANK_PROFILE_DESCRIPTOR::map);
     }
 
     @Transactional(readOnly = true)
-    public List<BankDescriptor> getBanks(String query, int startPos, int maxResults) {
+    public List<BankDescriptor> getBanks(String query, int startPos, int maxResults, boolean onlyActive) {
         return bankRepository.getBanks(query, startPos, maxResults)
                 .stream()
+                .filter(bank -> !onlyActive || bank.isActive())
```
Comment (valb3r) [inline opba-banking-rest-impl/src/main/java/de/adorsys/opba/tppbankingapi/service/BankService.java:36]: Use filtering directly in the query, as this kind of filtering breaks pagination (I.e. page that has all-inactive banks will yield empty bank response)

File: opba-banking-rest-api-ymls/src/main/resources/static/tpp_banking_api_bank_search.yml [hunk 2]
```diff
@@ -81,7 +83,18 @@ paths:
 components:
   headers: {}
 
-  parameters: {}
+  parameters:
+
```
Comment (valb3r) [inline opba-banking-rest-api-ymls/src/main/resources/static/tpp_banking_api_bank_search.yml:87]: Please remove blank line

File: opba-banking-rest-api-ymls/src/main/resources/static/tpp_banking_api_bank_search.yml [hunk 3]
```diff
@@ -81,7 +83,18 @@ paths:
...
+    onlyActive:
+      name: onlyActive
+      in: query
+      required: false
+      description: |
+        Makes possible to get not active item.
```
Comment (valb3r) [inline opba-banking-rest-api-ymls/src/main/resources/static/tpp_banking_api_bank_search.yml:93]: `Makes possible to get not active item.` -> `Controls visibility of inactive banks in the response (false - both active and inactive, true - only active)`

File: opba-banking-rest-api-ymls/src/main/resources/static/tpp_banking_api_bank_search.yml (hunk 2 above)
Comment (tsehelnyk) [reply opba-banking-rest-api-ymls/src/main/resources/static/tpp_banking_api_bank_search.yml:87]: done

File: opba-banking-rest-api-ymls/src/main/resources/static/tpp_banking_api_bank_search.yml (hunk 3 above)
Comment (tsehelnyk) [reply opba-banking-rest-api-ymls/src/main/resources/static/tpp_banking_api_bank_search.yml:93]: fixed

File: opba-banking-rest-impl/src/main/java/de/adorsys/opba/tppbankingapi/service/BankService.java (hunk 1 above)
Comment (tsehelnyk) [reply opba-banking-rest-impl/src/main/java/de/adorsys/opba/tppbankingapi/service/BankService.java]: filtering was moved to query

File: opba-banking-rest-impl/src/test/resources/application.yml [hunk 4]
```diff
@@ -28,14 +28,14 @@ api:
       start: 0
       max: 10
       query:
-        "SELECT id, uuid, name, bic, bank_code, word_similarity(:keyword, name) as sml
-        FROM opb_bank WHERE word_similarity(:keyword, name) >= 0.6
+        "SELECT id, uuid, name, bic, bank_code, active, word_similarity(:keyword, name) as sml
```
Comment (valb3r) [inline opba-banking-rest-impl/src/test/resources/application.yml:31]: Please replace query value in all YMLs (i.e. open-banking-embedded-starter)

Comment (tsehelnyk) [reply opba-banking-rest-impl/src/test/resources/application.yml:31]: replaced

**Review by valb3r — APPROVED** [review]

**Pull request merged**